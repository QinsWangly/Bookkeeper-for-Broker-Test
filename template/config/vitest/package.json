{
  "scripts": {
    "test:unit": "vitest"
  },
  "dependencies": {
    "vue": "^2.7.16"
  },
  "devDependencies": {
    "@vue/test-utils": "^1.3.6",
    "jsdom": "^24.0.0",
    "vitest": "^1.2.1",
    "vue-template-compiler": "^2.7.16"
  }
}
