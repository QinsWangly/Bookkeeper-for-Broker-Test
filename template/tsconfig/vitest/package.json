{
  "scripts": {
    "typecheck": "vue-tsc --noEmit -p tsconfig.json --composite false"
  },
  "devDependencies": {
    "@types/jsdom": "^16.2.14"
  }
}
