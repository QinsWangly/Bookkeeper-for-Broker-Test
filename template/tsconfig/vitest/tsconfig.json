{
  "files": [],
  "references": [
    {
      "path": "./tsconfig.vite-config.json"
    },
    {
      "path": "./tsconfig.vitest.json"
    },
    {
      "path": "./tsconfig.app.json"
    }
  ]
}
