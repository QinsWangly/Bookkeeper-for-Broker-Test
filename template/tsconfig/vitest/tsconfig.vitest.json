{
  "extends": "./tsconfig.app.json",
  "include": ["src/**/__tests__/*"],
  "exclude": [],
  "compilerOptions": {
    "composite": true,
    "lib": [],
    "types": ["node", "jsdom"]
  }
}
