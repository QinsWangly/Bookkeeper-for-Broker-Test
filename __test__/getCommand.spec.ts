import { it, describe, expect } from 'vitest'
import getCommand from '../utils/getCommand'

describe("should generate correct command", () => {
    it('for yarn', () => {
        expect(getCommand('yarn', 'install')).toBe('yarn')
        expect(getCommand('yarn', 'install')).toMatchSnapshot()
        expect(getCommand('yarn', 'dev')).toBe('yarn dev')
        expect(getCommand('yarn', 'dev')).toMatchSnapshot()
        expect(getCommand('yarn', 'build')).toBe('yarn build')
        expect(getCommand('yarn', 'build')).toMatchSnapshot()

    })
    it('for npm', () => {
        expect(getCommand('npm', 'install')).toBe('npm install')
        expect(getCommand('npm', 'install')).toMatchSnapshot()
        expect(getCommand('npm', 'dev')).toBe('npm run dev')
        expect(getCommand('npm', 'dev')).toMatchSnapshot()
        expect(getCommand('npm', 'build')).toBe('npm run build')
        expect(getCommand('npm', 'build')).toMatchSnapshot()

    })
    it('for pnpm', () => {
        expect(getCommand('pnpm', 'install')).toBe('pnpm install')
        expect(getCommand('pnpm', 'install')).toMatchSnapshot()
        expect(getCommand('pnpm', 'dev')).toBe('pnpm dev')
        expect(getCommand('pnpm', 'dev')).toMatchSnapshot()
        expect(getCommand('pnpm', 'build')).toBe('pnpm build')
        expect(getCommand('pnpm', 'build')).toMatchSnapshot()
    })
})