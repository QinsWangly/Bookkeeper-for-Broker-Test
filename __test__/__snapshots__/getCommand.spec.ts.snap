// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should generate correct command > for npm 1`] = `"npm install"`;

exports[`should generate correct command > for npm 2`] = `"npm run dev"`;

exports[`should generate correct command > for npm 3`] = `"npm run build"`;

exports[`should generate correct command > for pnpm 1`] = `"pnpm install"`;

exports[`should generate correct command > for pnpm 2`] = `"pnpm dev"`;

exports[`should generate correct command > for pnpm 3`] = `"pnpm build"`;

exports[`should generate correct command > for yarn 1`] = `"yarn"`;

exports[`should generate correct command > for yarn 2`] = `"yarn dev"`;

exports[`should generate correct command > for yarn 3`] = `"yarn build"`;
